tests/eas/cpuhp.py

```diff
@@ -0,0 +1,217 @@
+        'seed' : None,                  # Seed of the random number generator
+        'sequence_len' : 100,           # Number of operations in the sequence
+        'sleep' : {
+            'min_ms' : 10,              # Min sleep duration between hotplugs 
+            'max_ms' : 100,             # Max sleep duration between hotplugs
+        },                              #   max_ms <= 0 will encode 'no sleep'
+        'max_cpus_off' : sys.maxint,    # Max number of CPUs plugged-off
+    }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(_CpuHotplugTest, cls).runExperiments(*args, **kwargs)
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        conf = {
+            'tag' : 'energy_aware',
+            'flags' : ['ftrace', 'freeze_userspace'],
+            'sched_features' : 'ENERGY_AWARE',
+        }
+
+        return {
+            'wloads' : cls.workloads,
+            'confs' : [conf],
+        }
+
+    @classmethod
+    def _experimentsInit(cls, *args, **kwargs):
+        # calibration shoud be done before starting spurious hotplugs
+        if not cls.te._calib:
+            cls.te.calibration()
+        
+        # Choose a random seed explicitly if not given
+        if 'seed' not in cls.hp_stress or cls.hp_stress['seed'] is None:
+            random.seed() 
+            cls.hp_stress['seed'] = random.randint(0, sys.maxint)
+        random.seed(cls.hp_stress['seed'])
+        cls.te._log.info('Random sequence of CPU Hotplug generated with: ')
+        cls.te._log.info(cls.hp_stress)
+        with open(os.path.join(cls.te.res_dir, 'hotplug_cfg.json'), 'w') as f:
+            json.dump(cls.hp_stress, f, sort_keys=True, indent=4)
+       
+        # Play with (online) hotpluggable CPUs only
+        cls.hotpluggable_cpus = filter(
+                lambda x: cls.target.file_exists(hp._cpu_path(cls.target, x)),
+                cls.target.list_online_cpus())
+        if not cls.hotpluggable_cpus:
+            raise RuntimeError('Cannot find any hotpluggable CPU online')
+        cls.te._log.info('Hotpluggable CPUs found on target: ')
+        cls.te._log.info(cls.hotpluggable_cpus)
+
+        # Run random hotplug sequence on target
+        cpuhp_seq_script = cls._random_cpuhp_script()
+        cmd = cls.target.install(cpuhp_seq_script)
+        cls.hotplug_shell = cls.target.background(cmd, as_root=True)
+        cls.te._log.info('Hotplug stress has now started on target')
+    
+    @classmethod
+    def _experimentsFinalize(cls, *args, **kwargs):
+        cls.te._log.info('Stopping hotplug stress on the target now')
+        # Kill hotplug script
+        pid = cls.target.read_int(cls.pid_file)
+        cls.target.kill(pid, as_root=True)
+        cls.hotplug_shell.kill() 
+        # Resume to original state
+        cls.te._log.info('Plugging back in currently offline CPUs')
+        cls.target.hotplug.online(*cls.hotpluggable_cpus)
+
+    @classmethod
+    def _random_cpuhp_script(cls):
```

Comment:
This looks like a job for libs/util/target_script (which is easy for me to say since I wrote it). As per our offline chat, that class will need a few modifications for your test, so do feel free to make it better.